spec.md

```diff
@@ -441,9 +452,23 @@ A registry SHOULD enforce some limit on the maximum manifest size that it can ac
 A registry that enforces this limit SHOULD respond to a request to push a manifest over this limit with a response code `413 Payload Too Large`.
 Client and registry implementations SHOULD expect to be able to support manifest pushes of at least 4 megabytes.
 
+###### Pushing Manifests with Refers
+
+When pushing an Image or Artifact Manifest with the `refers` field and the [Referrers API](#listing-referrers) returns a 404, the client MUST:
+
+1. Pull the current Referrers list using the [referrers tag schema](#referrers-tag-schema).
+1. If that pull returns a Manifest other than the expected Image Index, the client SHOULD report a failure and skip the remaining steps.
+1. If the tag returns a 404, the client MUST begin with an empty Image Index.
+1. Verify the descriptor for the manifest is not already in the Referrers list (duplicate entries SHOULD NOT be created).
+1. Append a descriptor for the pushed Image or Artifact Manifest to the manifests in the Referrers list.
+   The value of the `artifactType` MUST be set in the descriptor to value of the `artifactType` in the Artifact Manifest, or the Config descriptor `mediaType` in the Image Manifest.
+   All annotations from the Image or Artifact Manifest MUST be copied to this descriptor.
```

Comment:
what if they update values, need to delete, or otherwise need to merge? Any advice for readers.